playground-app/src/App.tsx

```diff
@@ -222,19 +248,49 @@ function App() {
             }}
           >
             <h4>Feature definition</h4>
-            <textarea
-              style={{
-                width: "100%",
-                minHeight: "500px",
-                boxSizing: "border-box",
-                resize: "vertical",
-                padding: "8px",
-                ...codeStyle,
-              }}
-              name="feature-definition"
-              value={featureDefinition}
-              onChange={(e) => setFeatureDefinition(e.target.value)}
-            />
+            <div style={{ backgroundColor: codeStyle.backgroundColor }} >
+              <Editor
+                theme={editorTheme}
+                width="100%"
+                height="500px"
+                defaultLanguage="json"
+                value={featureDefinition}
+                options={{
+                  minimap: { enabled: false }
+                }}
+                beforeMount={(monaco) => {
+                  // inherent from the normal vs/vs-dark themes, but with transparent backgrounds
+                  // so our CSS variables can be used for that (css vars cant be used in the editor theme directly)
+                  monaco?.editor.defineTheme('custom-dark', {
+                    base: 'vs-dark', // inherent all the normal "dark" syntax highlighting
+                    inherit: true,
+                    rules: [],
+                    colors: {
+                      "editor.foreground": "#00000000",
+                      "editor.background": "#00000000",
+                    }
+                  });
+                  monaco?.editor.defineTheme('custom', {
+                    base: 'vs', // inherent all the normal "light" syntax highlighting
+                    inherit: true,
+                    rules: [],
+                    colors: {
+                      "editor.foreground": "#00000000",
+                      "editor.background": "#00000000",
+                    }
+                  });
+                  monaco?.languages.json.jsonDefaults.setDiagnosticsOptions({
+                    enableSchemaRequest: true,
+                    allowComments: true,
```

Comment:
@beeme1mr @Kavindu-Dodan I wasn't sure about this. This allows comments in JSON, which is nice for educational purposes and even renders appropriately in monaco, but will cause errors if they are copy/pasted into flagd, obviously.